src/libsystemd-network/network-internal.c

```diff
@@ -86,6 +86,32 @@ int net_get_unique_predictable_data(struct udev_device *device, uint64_t *result
         return 0;
 }
 
+static bool net_condition_test_strv(char * const *raw_patterns, char *string) {
```

Comment:
hmm, the function returns a "bool" but internally allocates memory, which can fail. Thus OOM might be considered equal to a "false" test, which I think isn't right.

I think the test can be changed relatively easily to not require heap allocation though. Note that "strv" objects are little more than NULL terminate arrays of char*. Hence, let's simply prep our own copy of one, but on the stack:

```
unsigned n, i;
char **copy;

n = strv_length(patterns)
copy = newa(char*, n+1);
copy[0] = patterns[0] + 1; /* skip the ~ */
for (i = 1; i < n+1; i++)
        copy[i] = patterns[i];
…
```

This way we don't duplicate the full strv, just the array. And since we don't need it after the function exits, we are fine... No heap allocation, just stack allocation of the array, and no need to handle malloc() failed...